### DISCUSSION TITLE
Add global callbacks for all dags

### DISCUSSION THREAD

**Discussion by yupbank** [apache/airflow#31621] [Ideas]
### Description

I find it's useful to add global on_success and on_failure callbacks for all dags.  and not being able to find a way to do that.

### Use case/motivation

example, some kind of plugin can allow user to setup some callback for all existing DAGS without the need to updating them

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by arjunanan6** [comment]
A feature like this would be great. What we've done for now is to add our re-usable modules (such as on_success or on_failure callbacks) as [plugins ](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/plugins.html)and simply import them into our DAGs. Granted, you still need to import it for each DAG, but it makes the effort a little easier.

**Comment by hussein-awala** [comment]
You can use Airflow Listeners to create TIs/DAG runs lifecycle notifications, and they are applied on all the dags.
https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/listeners.html
